functions.php

```diff
@@ -905,6 +911,84 @@ public function override_img_caption_shortcode( $output, $attr, $content ) {
 		return $output;
 	}
 
+	/**
+     * Add extra date column in post rest endpoint.
+     *
+	 * @param array $args Arguments array passed to endpoint.
+	 * @param array $request Initial request parameters to the endpoint.
+	 *
+	 * @return mixed
+	 */
+	function add_posts_param_to_endpoint( $args, $request ) {
+		if ( ! isset( $request['before'] ) && ! isset( $request['after'] ) ) {
+			return $args;
+		}
+
+		if ( isset( $request['date_query_column'] ) ) {
+			$args['date_query'][0]['column'] = $request['date_query_column'];
+		}
+
+		return $args;
+	}
+
+	/**
+     * Use date_query_column added in post rest endpoint in order to query on posts/pages modified attribute.
+     *
+	 * @param array $query_params Rest endpoint query parameters.
+	 *
+	 * @return mixed
+	 */
+	function filter_post_params_endpoint( $query_params ) {
+		$query_params['date_query_column'] = [
+			'description' => __( 'The date query column.' ),
+			'type'        => 'string',
+			'enum'        => [ 'post_date', 'post_date_gmt', 'post_modified', 'post_modified_gmt' ],
+		];
+
+		return $query_params;
+	}
+
+	/**
+	 * Add post report submenu item.
+	 */
+	function add_posts_report_admin_menu_item() {
+		add_posts_page(
+			__( 'Posts Report', 'planet4-master-theme' ),
+			__( 'Posts Report', 'planet4-master-theme' ),
+			'read',
+			'posts-report',
+			[ $this, 'render_posts_report_page' ]
+		);
+	}
+
+	/**
+	 * Callback function to render posts report page.
+	 */
+	public function render_posts_report_page() {
+		wp_enqueue_script( 'jquery-ui-core' );
+		wp_enqueue_script( 'jquery-ui-datepicker' );
+		wp_register_script( 'posts-report', $this->theme_dir . '/assets/js/posts_report.js',
+			[
+				'jquery',
+				'wp-api',
+				'wp-backbone',
+			], '0.1', true );
+		wp_localize_script(
+			'posts-report',
+			'p4_data',
+			[
+				'api_url' => '/wp-json/wp/v2',
+				'nonce'   => wp_create_nonce( 'wp_rest' ),
+			]
+		);
+		wp_enqueue_script( 'posts-report' );
+		wp_register_style(
+			'jquery-ui-datepicker',
+			'https://cdnjs.cloudflare.com/ajax/libs/jqueryui/1.12.1/themes/overcast/jquery-ui.min.css'
+		);
+		wp_enqueue_style( 'jquery-ui-datepicker' );
+		include 'posts_report.php';
+	}
```

Comment:
@kirdia What do you think about moving the functionality about the post's report page to its own class under the classes directory as we have done in other cases? This way we can keep the functions.php as small as possible and the P4_Master_Site class without too many concerns.

Reply:
Yes I agree, I'll move the functionality to a new class.